src/node/ledger_secrets.h

```diff
@@ -124,22 +91,13 @@ namespace ccf
       // require access to a tx object, which must take a dependency on the
       // secrets table.
       auto secrets = tx.ro<Secrets>(Tables::ENCRYPTED_LEDGER_SECRETS);
-
-      // Taking a read dependency on the key at self, which would get updated
-      // on rekey
-      if (!self.has_value())
-      {
-        throw std::logic_error(
-          "Node id should be set before taking dependency on secrets table");
-      }
-      secrets->get(self.value());
+      secrets->get(0);
```

Comment:
I'm unable to grok how this relates to the rest of the PR - why do we no longer store secrets per-node?

Reply:
We still do. The previous schema of the `secrets` table was `NodeId` to `EncryptedLedgerSecrets`. This meant two things: `LedgerSecrets` had to know their node ID and the table was forever growing, even when will we delete retired nodes from the store. Passing the node ID to the historical ledger secrets was awkward so I've changed the schema for the table so that we only store secrets at key `0`.